lib/Controller/SettingsController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2023 Nextcloud GmbH
+ *
+ * @author Ferdinand Thiessen <opensource@fthiessen.de>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\LogReader\Controller;
+
+use OCA\LogReader\Constants;
+use OCA\LogReader\Service\SettingsService;
+use OCP\AppFramework\ApiController;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IConfig;
+use OCP\IRequest;
+
+use Psr\Log\LoggerInterface;
+
+class SettingsController extends ApiController {
+
+	public function __construct(string $appName,
+		private SettingsService $settingsService,
+		private IConfig $config,
+		private LoggerInterface $logger,
+		IRequest $request) {
```

Comment:
```suggestion
	public function __construct(
		string $appName,
		private SettingsService $settingsService,
		private IConfig $config,
		private LoggerInterface $logger,
		IRequest $request,
	) {
```
Also if possible move $request parameter to before the private properties